openquakeplatform_ipt/static/ipt/js/earthquake_rupture.js

```diff
@@ -0,0 +1,626 @@
+     ************/
+    planar_surface_del: function (obj) {
+        var id = obj.getAttribute("data_gem_id");
+        var item = $(er_obj.pfx + 'div[name="planars"] div[name="planar-' + id + '"]');
+        delete(this.planar_tbl[id]);
+        item.remove();
+    },
+
+    planar_surface_add: function () {
+        var ct = er_obj.planar_tbl_cur;
+        var ctx = '\
+      <div name="planar-' + ct + '">\n\
+        <div class="menuItems" style="margin-top: 12px;">\n\
+<div style="display: inline-block; float: left;"><h4>Planar surface ' + (ct + 1) + '</h4></div>';
+        ctx += (ct == 0 ? '<div style="clear: both;"></div>' : '<button type="button" data_gem_id="' + ct + '" class="btn" style="margin-top: 8px; margin-bottom: 8px;" onclick="er_obj.planar_surface_del(this);">Delete Planar Surface</button>');
+        ctx += '\
+        </div>\n\
+        <div class="menuItems">\n\
+         <label>Strike (degrees) <span class="ui-icon ui-icon-help ipt_help" title="The strike direction corresponds to the angle between the north and the direction you take so that when you walk along the fault trace the fault dips on your right."></span>:</label>\n\
+          <input type="text" name="strike" value="0" placeholder="0 ≤ float ≤ 360">\n\
+        </div>\n\
+        <div class="menuItems">\n\
+          <label>Dip (degrees)  <span class="ui-icon ui-icon-help ipt_help" title="The dip is the steepest angle of descent of the fault plane relative to a horizontal plane; it is measured in degrees (0, 90].">:</label>\n\
+          <input type="text" name="dip" placeholder="0 < float ≤ 90" value="90">\n\
+        </div>\n\
+\n\
+        <div class="menuItems planar_geometry">\n\
+          <label>Planar Geometry:</label>\n\
+          <div style="margin-left: auto;">\n\
+            <div name="geometry-' + ct + '" style="margin-left: auto; width: 340px; height: 140px; overflow: hidden;"></div>\n\
+          </div>\n\
+        </div>\n\
+</div>';
+        $(er_obj.pfx + 'div[name="planars"]').append(ctx);
+
+        var table_id = er_obj.pfx + 'div[name="rupture"] > div[name="planar"] div[name="geometry-' + ct + '"]';
+
+        $(table_id).handsontable({
+            colHeaders: [ 'Longitude (°)', 'Latitude (°)', 'Depth (km)'],
+            rowHeaders: ["topLeft", "topRight", "bottomLeft", "bottomRight"],
+            startCols: 3,
+            startRows: 4,
+            maxCols: 3,
+            maxRows: 4,
+            className: "htRight"
+        });
+        er_obj.planar_tbl[ct] = $(table_id).handsontable('getInstance');
+
+        er_obj.planar_tbl_cur++;
+    },
+
+    /*************
+     *           *
+     *  COMPLEX  *
+     *           *
+     *************/
+    complex_surface_middle_del: function (obj) {
+        var id = obj.getAttribute("data_gem_id");
+        var mid_id = obj.getAttribute("data_gem_mid_id");
+        var $item = $(er_obj.pfx + 'div[name="complexes"] div[name="complex-' + id + '"] div[name="middle-' + mid_id + '"]');
```

Comment:
$item with '$' just to keep trace that is a JQuery filtering result and not just a dom object or a simple variable.